src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx

```diff
@@ -30,31 +35,97 @@ export const SelectTokenFromListScreen = () => {
   })
 
   const wallet = useSelectedWallet()
-  const balances = useBalances(wallet)
 
-  const tokenInfos = useTokenInfos({
-    wallet,
-    tokenIds: Amounts.toArray(balances).map(({tokenId}) => tokenId),
-  })
   const secondaryAmountsCounter = useSelectedSecondaryAmountsCounter(wallet)
-  const canAddAmount = secondaryAmountsCounter < limitOfSecondaryAmountsPerTx
 
-  const {search: assetSearchTerm} = useSearch()
-  const sortedTokenInfos = sortTokenInfos({wallet, tokenInfos: filterAssets(assetSearchTerm, tokenInfos)})
-  const isSearchResultEmpty = assetSearchTerm.length > 0 && sortedTokenInfos.length === 0
+  const {visible: isSearching} = useSearch()
+
+  const canAddAmount = secondaryAmountsCounter < limitOfSecondaryAmountsPerTx
 
   return (
     <View style={styles.root}>
-      {!canAddAmount && (
-        <View style={styles.panel}>
-          <MaxAmountsPerTx />
+      <View style={styles.subheader}>
+        {!isSearching && (
+          <Tabs>
+            <Tab active={fungibilityFilter} onPress={setFungibilityFilter} label={strings.all} tab="all" />
+
+            <Tab active={fungibilityFilter} onPress={setFungibilityFilter} label={strings.tokens(2)} tab="ft" />
+
+            <Tab active={fungibilityFilter} onPress={setFungibilityFilter} label={strings.nfts(2)} tab="nft" />
+          </Tabs>
+        )}
+
+        {!canAddAmount && (
+          <View style={styles.panel}>
+            <MaxAmountsPerTx />
+
+            <Spacer height={16} />
+          </View>
+        )}
+      </View>
+
+      <List fungibilityFilter={fungibilityFilter} isSearching={isSearching} canAddAmount={canAddAmount} />
+    </View>
+  )
+}
+
+type ListProps = {
+  fungibilityFilter: FungibilityFilter
+  isSearching: boolean
+  canAddAmount: boolean
+}
+
+const List = ({fungibilityFilter, isSearching, canAddAmount}: ListProps) => {
+  const isNftListVisible = fungibilityFilter === 'nft' && !isSearching
+
+  if (isNftListVisible) return <NftList />
+  return <AssetList fungibilityFilter={fungibilityFilter} canAddAmount={canAddAmount} />
+}
+
+const NftList = () => {
+  const wallet = useSelectedWallet()
+  const navigation = useNavigation<TxHistoryRouteNavigation>()
+  const {tokenSelectedChanged, amountChanged} = useSend()
+  const balances = useBalances(wallet)
+
+  const {nfts} = useNfts(wallet)
+  const sortedNfts = nfts.sort((NftA, NftB) => sortNfts(NftA.name, NftB.name))
 
-          <Spacer height={16} />
-        </View>
-      )}
+  const onSelect = (nftId) => {
```

Comment:
@jemubm both onSelect, here and in the asset, need to reset the search, otherwise when navigating back the search will be open (clearSearch + hideSearch)

Reply:
maybe we can combine into a new action `closeSearch` -> {search: '', visible: false}

Reply:
so we dispatch only 1 action.